src/err/mod.rs

```diff
@@ -443,15 +467,12 @@ impl PyErr {
     /// This is the opposite of `PyErr::fetch()`.
     #[inline]
     pub fn restore(self, py: Python<'_>) {
-        let state = match self.state.into_inner() {
-            Some(state) => state,
+        match self.state.into_inner() {
+            Some(state) => state.restore(py),
             // Safety: restore takes `self` by value so nothing else is accessing this err
             // and the invariant is that state is always defined except during make_normalized
             None => unsafe { std::hint::unreachable_unchecked() },
```

Comment:
I know it is unrelated but the safety comment seems out of date as there is no `make_normalized` function any more? Asked more broadly, is the `unreachable_unchecked` performance critical here or would the simpler `expect("PyErr state should never be invalid outside of normalization")` suffice as well?

Reply:
Benchmarks indicated no difference with the `unreachable_unchecked`, so let's go with expect. 👍